core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2608,8 +2608,12 @@ private[spark] class DAGScheduler(
     // isPushBasedShuffleEnabled and shuffleMergers need to be updated at the end.
     stage match {
       case s: ShuffleMapStage =>
-        stage.latestInfo.setPushBasedShuffleEnabled(s.shuffleDep.getMergerLocs.nonEmpty)
-        stage.latestInfo.setShuffleMergerCount(s.shuffleDep.getMergerLocs.size)
+        val isPushShuffleEnabled =
+          s.shuffleDep.shuffleMergeAllowed && s.shuffleDep.getMergerLocs.nonEmpty
```

Comment:
When will we have the case where mergerLocs is non empty, but merge is not allowed ?

Reply:
Yes exactly. That makes sense.